sdk/iot_hub/src/service/requests/delete_configuration.rs

```diff
@@ -0,0 +1,37 @@
+use crate::service::{ServiceClient, API_VERSION};
+
+use azure_core::headers;
+use azure_core::Method;
+
+azure_core::operation! {
+    /// The DeleteConfigurationBuilder is used to construct a request to delete a configuration.
+    DeleteConfiguration,
+    client: ServiceClient,
+    if_match: String,
```

Comment:
Similar with above, we should standardize on `etag` vs `if_match`.  Additionally, this should be optional.

https://docs.microsoft.com/en-us/rest/api/iothub/service/configuration/delete#request-headers